# Project Planning Individual Stage

In [1]:
### Run this cell before continuing. 
library(tidyverse)
library(repr)
options(repr.matrix.max.rows = 6)
source("cleanup.R")

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Warning message in file(filename, "r", encoding = encoding):
“cannot open file 'cleanup.R': No such file or directory”


ERROR: Error in file(filename, "r", encoding = encoding): cannot open the connection


## 1. Data Description

### players.csv

This data shows the identities of players and their experiences and behaviours in the game. The data has a total of **196 observations** and **7 variables**.
The variables are as listed below:

| **Variable**     | **Type** | **Description**                     | **Notes** |
|------------------|----------|--------------------------------------|-----------|
| `experience`     | chr      | Players gaming experience level (Beginner, Amateur, Regular, Veteran, Pro) | Might be subjectively classified. |
| `subscribe` | lgl | Whether each player subscribed (TRUE) to a game-related newsletter or not (FALSE)  | Can be used as a target variable for prediction |
| `hashedEmail` | chr | Player ID | 196 different IDs. Can act as an identification for each player |
| `played_hours` | dbl | Number of hours played by each player | Might be right-skewed with some invalid data |
| `name` | chr | The name of each player | 196 different names. Can act as the same function as `hashedEmail` | 
| `gender` | chr | The gender of each player | Mostly male/female. Some group of gender might be  |
| `Age` | dbl | The age of each player | Have two missing values. Can be used as a predictor. |

### Potential Data Issues

- Two missing values in `Age` can lead to error in data processing
- The number in `played_hours` is might be right-skewed (many low-hour players, few high-hour players)
- Some values in the `played_hours` column is 0.0, which means the player played for 0.0 hours, and can be invalid.
- `gender` column is dominated by `Male` and `Female`. Therefore, other assigned genders might be underrepresented in the data processing.
- The values in the `experience` column might be subjective since there is not formal definition of each gaming experiences.

### Likely Data Collection Method

- `subscribe` data is collected from the in-app subscription data.
- `experience`, `name`, `gender`, and `Age` data is likely to be user-input data.
- `played_hours` likely comes from platform usage logs.

### Summary Statistics

In [3]:
players <- read_csv("data_individual/players.csv")

players |> summarise(
    played_hours_mean = mean(played_hours, na.rm=TRUE),
    age_mean = mean(Age, na.rm = TRUE),
    played_hours_median = median(played_hours, na.rm=TRUE),
    total_players = n(),
    subscribe_count = sum(subscribe == "TRUE", na.rm=TRUE),
    subscribe_percentage = subscribe_count / total_players *100)|>
    round(2)


Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


played_hours_mean age_mean played_hours_median total_players subscribe_count
1 5.85              21.14    0.1                 196           144            
  subscribe_percentage
1 73.47

- Mean Played Hours = 5.85 hours
- Mean Age = 21.14
- Median Played Hours = 0.1 hours
- Total Players = 196 people
- Count Subscribe = 144
- Percentage of Players subscribed to newsletter = 73.47%

## 2. Question

### Broad Question: 
What player characteristics and behaviours are most predictive of subscribing to a game-related newsletter, and how do these features differ between various player types?

### Specific Question:
Can the player's age, total gameplay time, and game experience predict if a player is subscribed to a game-related newsletter or not in `players.csv`?

## 3. Exploratory Data Analysis and Visualization

In [4]:
players

experience subscribe
1   Pro         TRUE    
2   Veteran     TRUE    
3   Veteran    FALSE    
⋮   ⋮          ⋮        
194 Amateur    FALSE    
195 Amateur    FALSE    
196 Pro         TRUE    
    hashedEmail                                                     
1   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
2   f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9
3   b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28
⋮   ⋮                                                               
194 d572f391d452b76ea2d7e5e53a3d38bfd7499c7399db299bd4fedb06a46ad5bb
195 f19e136ddde68f365afc860c725ccff54307dedd13968e896a9f890c40aea436
196 d9473710057f7d42f36570f0be83817a4eea614029ff90cf50d8889cdd729d11
    played_hours name      gender            Age
1   30.3         Morgan    Male               9 
2    3.8         Christian Male              17 
3    0.0         Blake     Male              17 
⋮   ⋮            ⋮         ⋮                 ⋮  
194 0.0          Dylan     Prefer not to say 57 
195 2.3          Harlow    Male              17 
196 0.2          Ahmed     Other             NA